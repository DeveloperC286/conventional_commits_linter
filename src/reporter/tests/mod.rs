use rstest::rstest;

use super::*;

#[rstest(
    message,
    snapshot_name,
    case(
        "feat()!: yargs-parser now throws on invalid combinations of config\n\n",
        "test_stdin_pretty_print_case_1"
    ),
    case("Release 1.0.126", "test_stdin_pretty_print_case_2"),
    case(
        "Merge branch 'fix-ocsp-signer-check' into 'master'",
        "test_stdin_pretty_print_case_3"
    )
)]
fn test_stdin_pretty_print(message: &str, snapshot_name: &str) {
    // Given
    let commit = Commit {
        oid: git2::Oid::zero(),
        message: message.to_string(),
    };
    let linting_errors = crate::linter::lint_commit(&commit, false);

    // When
    let reporting = pretty_print_linting_error(None, message, &linting_errors);

    // Then
    insta::assert_snapshot!(snapshot_name, reporting);
}
