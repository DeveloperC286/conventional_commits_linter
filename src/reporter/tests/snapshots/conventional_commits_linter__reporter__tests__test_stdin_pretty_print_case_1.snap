---
source: src/reporter/tests/mod.rs
expression: reporting

---
[1;31mMessage[0m - "feat()!: yargs-parser now throws on invalid combinations of config\n\n"
	[1;31mX[0m - Commit title does not comply with the Conventional Commits V1.0.0 specification.
	[1;31mX[0m - Commit title has a scope which is empty.


